satpy/composites/lookup.py

```diff
@@ -0,0 +1,208 @@
+# Copyright (c) 2015-2025 Satpy developers
+#
+# This file is part of satpy.
+#
+# satpy is free software: you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# satpy is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# satpy.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Compositors using lookup tables."""
+
+from __future__ import annotations
+
+import logging
+
+import numpy as np
+import xarray as xr
+
+from .core import CompositeBase, GenericCompositor
+
+LOG = logging.getLogger(__name__)
+
+
+class CategoricalDataCompositor(CompositeBase):
+    """Compositor used to recategorize categorical data using a look-up-table.
+
+    Each value in the data array will be recategorized to a new category defined in
+    the look-up-table using the original value as an index for that look-up-table.
+
+    Example:
+        data = [[1, 3, 2], [4, 2, 0]]
+        lut = [10, 20, 30, 40, 50]
+        res = [[20, 40, 30], [50, 30, 10]]
+    """
+
+    def __init__(self, name, lut=None, **kwargs):  # noqa: D417
+        """Get look-up-table used to recategorize data.
+
+        Args:
+            lut (list): a list of new categories. The lenght must be greater than the
+                        maximum value in the data array that should be recategorized.
+        """
+        self.lut = np.array(lut)
+        super(CategoricalDataCompositor, self).__init__(name, **kwargs)
+
+    def _update_attrs(self, new_attrs):
+        """Modify name and add LUT."""
+        new_attrs["name"] = self.attrs["name"]
+        new_attrs["composite_lut"] = list(self.lut)
+
+    @staticmethod
+    def _getitem(block, lut):
+        return lut[block]
+
+    def __call__(self, projectables, **kwargs):
+        """Recategorize the data."""
+        if len(projectables) != 1:
+            raise ValueError("Can't have more than one dataset for a categorical data composite")
+
+        data = projectables[0].astype(int)
+        res = data.data.map_blocks(self._getitem, self.lut, dtype=self.lut.dtype)
+
+        new_attrs = data.attrs.copy()
+        self._update_attrs(new_attrs)
+
+        return xr.DataArray(res, dims=data.dims, attrs=new_attrs, coords=data.coords)
+
+
+class ColormapCompositor(GenericCompositor):
```

Comment:
Wow this module name makes a lot more sense now that I realize that this compositor and the others below it are all deprecated.